sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -842,18 +843,37 @@ public String getAccountName() {
         return new SimpleResponse<>(response, directorySetMetadataInfo);
     }
 
-    private List<FileReference> convertResponseAndGetNumOfResults(
+    private List<StorageFileItem> convertResponseAndGetNumOfResults(
         DirectorysListFilesAndDirectoriesSegmentResponse response) {
-        Set<FileReference> fileReferences = new TreeSet<>(Comparator.comparing(FileReference::getName));
+        Set<StorageFileItem> storageFileItems = new TreeSet<>(Comparator.comparing(StorageFileItem::getName));
         if (response.getValue().getSegment() != null) {
             response.getValue().getSegment().getDirectoryItems()
-                .forEach(directoryItem -> fileReferences.add(new FileReference(directoryItem.getName(),
+                .forEach(directoryItem -> storageFileItems.add(new StorageFileItem(directoryItem.getName(),
                     true, null)));
             response.getValue().getSegment().getFileItems()
-                .forEach(fileItem -> fileReferences.add(new FileReference(fileItem.getName(), false,
+                .forEach(fileItem -> storageFileItems.add(new StorageFileItem(fileItem.getName(), false,
                     fileItem.getProperties())));
         }
 
-        return new ArrayList<>(fileReferences);
+        return new ArrayList<>(storageFileItems);
+    }
+
+    /**
```

Comment:
Can this method be made package private so you dont have to repeat the code in File and Directory?

Reply:
If make it package private, then we need to have an object to make the call. It is a little overhead, as DirectoryAsyncClient and FileAsyncClient does not have the other client initialized.
So better not to make it package private.

Reply:
The other option is having a package private static method which won't need to initialization overhead and would serve the same purpose.

Reply:
Do you mean we have another StorageFileUtility under File?

Reply:
Prefer to leave it here and abstract these if we have more helper methods needed. As this is internal changes, it will introduce any breaking changes.